### PR TITLE
Fix zookeeper connection leak

### PR DESCRIPTION
**Pull request by shanth96**
## Description

Fixes https://github.com/Shopify/vitess-project/issues/696

This PR closes the connection to the topo server whenever we get disconnected from zookeeper. This is to fix the connection/memory leak as described in the issue. It also adds additional logging whenever the connection is closed

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/topo/zk2topo/zk_conn_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	"vitess.io/vitess/go/testfiles"
+	"vitess.io/vitess/go/vt/zkctl"
+)
+
+func TestZkConnClosedOnDisconnect(t *testing.T) {
+	zkd, serverAddr := zkctl.StartLocalZk(testfiles.GoVtTopoZk2topoZkID, testfiles.GoVtTopoZk2topoPort)
+	defer zkd.Teardown()
+
+	conn := Connect(serverAddr)
```
Comment (brendar) [inline go/vt/topo/zk2topo/zk_conn_test.go:34]: Add `defer conn.Close()` to make sure the test cleans up after itself

File: go/vt/topo/zk2topo/zk_conn_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	_, _, err = conn.Get(context.Background(), "/")
+	if err != nil {
+		t.Fatalf("Get() failed: %v", err)
+	}
+
+	// Check that old connection is closed
+	_, _, err = oldConn.Get("/")
+	if err == nil {
+		t.Fatalf("Get() should have failed: %v", err)
+	}
```
Comment (brendar) [inline go/vt/topo/zk2topo/zk_conn_test.go]: Can replace this with `require.ErrorContains(t, err, "zookeeper is closing")` to ensure we got the expected error

Comment (shanth96) [reply go/vt/topo/zk2topo/zk_conn_test.go]: I think the error would be something like: `zk: could not connect to a server` so updated to check for that.

Comment (shanth96) [reply go/vt/topo/zk2topo/zk_conn_test.go]: Oh wait I see both `zk: zookeeper is closing` and `zk: could not connect to a server` from the logs in staging. I guess it depends on where exactly it fails?

**Review by brendar — APPROVED** [review]
🎉

**Pull request merged**